`byacc` causes `frameworks: 8 flex` to fail

**Describe the bug**

I do see this failure:
```
======================================== frameworks: 8 flex ========================================
 

Failed during: build
Reason: Compiling source code failed.
 

(inprocess) $ setup --prefix /usr --libdir lib 'test cases/frameworks/8 flex' '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/b 7f769b9486' --backend=ninja
The Meson build system
Version: 1.11.1
Source dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/frameworks/8 flex
Build dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/b 7f769b9486
Build type: native build
Project name: flex and bison
Project version: undefined
C compiler for the host machine: /usr/gcc/14/bin/gcc (gcc 14.3.0 "gcc (OpenIndiana 14.3.0-oi-1) 14.3.0")
C linker for the host machine: /usr/gcc/14/bin/gcc ld.solaris 5.11-1.1790 (illumos)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program flex found: YES (/usr/bin/flex)
Program bison found: YES (/usr/bin/bison)
test cases/frameworks/8 flex/meson.build:23: WARNING: Module codegen has no backwards or forwards compatibility and might not exist in future releases.
Program flex found: YES (/usr/bin/flex)
Message: lex implementation: flex
Program byacc found: YES (/usr/bin/byacc)
Message: yacc implementation: byacc
Build targets in project: 3
NOTICE: Future-deprecated features used:
 * 1.11.0: {'should_fail arg in test'}
WARNING: Deprecated features used:
 * 0.48.0: {'module python3'}
 * 0.55.0: {'Program.path'}
 * 0.56.0: {'dependency.get_configtool_variable', 'dependency.get_pkgconfig_variable', 'meson.source_root'}
 * 0.58.0: {'meson.get_cross_property'}
WARNING: Broken features used:
 * 1.10.1: {'Using '' as an empty value in link_args'}
 * 1.11.1: {'Using '' as an empty value in dependencies'}

flex and bison undefined

  User defined options
    backend: ninja
    libdir : lib
    prefix : /usr

Found ninja-1.13.2 at /usr/bin/ninja
NOTICE: You are using Python 3.9 which is EOL. Starting with v1.12.0, Meson will require Python 3.10 or newer
ninja explain: output parser.c doesn't exist
[1/6] Generating parser codegen-yacc-parser-build with byacc
FAILED: [code=1] parser.c parser.tab.h 
/usr/bin/byacc '../test cases/frameworks/8 flex/parser.y' -o parser.c -H parser.tab.h
Usage: /usr/bin/byacc [options] filename

Options:
  -b file_prefix        set filename prefix (default "y.")
  -B                    create a backtracking parser
  -d                    write definitions (.tab.h)
  -h                    print this help-message
  -H defines_file       write definitions to defines_file
  -i                    write interface (y.tab.i)
  -g                    write a graphical description
  -l                    suppress #line directives
  -L                    enable position processing, e.g., "%locations"
  -o output_file        (default ".tab.c")
  -p symbol_prefix      set symbol prefix (default "yy")
  -P                    create a reentrant parser, e.g., "%pure-parser"
  -r                    produce separate code and table files (y.code.c)
  -s                    suppress #define's for quoted names in %token lines
  -t                    add debugging support
  -v                    write description (y.output)
  -V                    show version information and exit

Long options:
  --defines             -H
  --file-prefix         -b
  --graph               -g
  --help                -h
  --name-prefix         -p
  --no-lines            -l
  --output              -o
  --version             -V
ninja explain: output lexer.c doesn't exist
[2/6] Generating lexer codegen-lex-lexer-build with flex
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Run `meson` tests on OpenIndiana.

**Expected behavior**
The test pass.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? native build
* operating system: OpenIndiana
* Python version: 3.9.25
* meson version: 1.11.1
* ninja version: 1.13.2
* byacc version: 2.0 20241231

--- a/mesonbuild/modules/codegen.py
+++ b/mesonbuild/modules/codegen.py
@@ -411,7 +411,7 @@ class CodeGenModule(ExtensionModule):
                     node=state.current_node)
             return YaccGenerator(name, bin)
 
-        yacc_args: T.List[str] = ['@INPUT@', '-o', '@OUTPUT0@']
+        yacc_args: T.List[str] = []
 
         impl = T.cast('YaccImpls', bin.name)
         if impl == 'yacc' and isinstance(bin, ExternalProgram):
@@ -433,6 +433,7 @@ class CodeGenModule(ExtensionModule):
                          'output location be configurable, and may not work.',
                          fatal=False)
             yacc_args.append('-H')
+        yacc_args.extend(['-o', '@OUTPUT0@', '@INPUT@'])
         return YaccGenerator(name, bin, T.cast('ImmutableListProtocol[str]', yacc_args))
 
 

